### DISCUSSION TITLE
Powering off and/or locking the display touch screen of the Spa using Wi-Fi remote?

### DISCUSSION THREAD

**Discussion by LateWiksi** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#582] [Q&A] [answered]
Hi,

Just got a Wi-Fi module on Friday and so far I've been happy with it and I have spent some time reverse engineering the WebUI and figuring out how I could control the Spa unit over Linux CLI. 

I figured out that the data can be read from the device using Arduino port 81 http://layzspa.local:81 and I was able to get out the settings in JSON format. I did as well find out that you can actually use curl and HTTP POST to activate commands via http://layzspa.local/addcommad/.

However I have been trying to figure out if there is a way to apply the display button lock on using Wi-Fi module as well as completely powering of the Spa unit?

You can see the button status in the States JSON
- PWR = Power status true/false
- LCK = Lock status true/false

`{
  "CONTENT": "STATES",
  "TIME": 1691257804,
  "LCK": 1,
  "PWR": 1,
  "UNT": 1,
  "AIR": 0,
  "GRN": 0,
  "RED": 0,
  "FLT": 1,
  "CH1": 32,
  "CH2": 50,
  "CH3": 50,
  "HJT": 0,
  "BRT": 7,
  "ERR": 0,
  "GOD": false,
  "TGT": 39,
  "TMP": 22,
  "VTMC": 21.98410416,
  "VTMF": 71.57138824,
  "AMBC": 21,
  "AMBF": 70,
  "AMB": 21,
  "VTM": 21.98410416,
  "TGTC": 39,
  "TMPC": 22,
  "TGTF": 102,
  "TMPF": 72
}`

Anyway, I did not find the cmd number for either power of lock. Wi-Fi hack contains index.js file that describes the cmd ID's for other comands.

index.js
`// command mapping
const cmdMap = {
	setTarget: 0,
	setTargetSelector: 0,
	toggleUnit: 1,
	toggleBubbles: 2,
	toggleHeater: 3,
	togglePump: 4,
	//resetq: 5,
	restartEsp: 6,
	//gettarget: 7,
	resetTotals: 8,
	resetTimerChlorine: 9,
	resetTimerFilter: 10,
	toggleHydroJets: 11,
	setBrightness: 12,
	setBrightnessSelector: 12,
	setBeep: 13,
	setAmbient: 15,
	setAmbientSelector: 15,
	setAmbientF: 14,
	setAmbientC: 15,
    resetDaily: 16,
    toggleGodmode: 17,
    setFullpower:18,
    printText: 19,
    setReady: 20
};`

E.g. from here I got the cmd ID for pump management. You can turn on the pump/filtering simply like in bellow. Cmd ID 4 for "togglePump" and value true means turn it on.
`curl -s -X POST -H "Content-Type: application/json" -d '{"CMD":"4","VALUE":true,"XTIME":'$(( $(date +%s) ))',"INTERVAL":0,"TXT":""}' http://layzspa.local/addcommand/`

So does anyone know is there a way for powering off or lock the display touch screen of the device?

The Spa it self is Lay-Z-Spa Miami model 2021.

Thanks in advance!

Regards
Wiksi

**Comment by visualapproach** [comment, marked as answer]
Hi, the commands are also described in the manual under mqtt. The 'power' button is, AFAIK, nothing more than dimming the display and lock the buttons. You can do that by setting brightness to zero and the display will auto lock when no input is made. What I'm saying is that the so called power function is not useful, and the lock function is mostly annoying. Therefore the esp automatically conveniently unlocks and power up the display when pressing buttons or sending commands. If you really want to prevent children or guests (invited or not) from messing with the pump, you can disable the display buttons from the UI (maybe it hasn't reached master branch yet but it's in the beta version)
- [x] You can even disable selected buttons. I disabled my timer button (it's horrible anyways) so pressing it wouldn't do anything to the pump/display but it still shows in 'pressed button' message. So my HA opened or closed the awnings instead :)

**Reply by LateWiksi** [reply]
Double checked the manual and found the command list. Thanks you!

I do agree that the lock feature is annoying, but as you described a few use cases, sometimes it might be useful. I'll looking to that disabling buttons. Disabling timer button is definitely a thing that I would also want to do.   


I will try that disabling buttons. ``

I check the manual again and found the

**Reply by Pilot001** [reply]
> The 'power' button is, AFAIK, nothing more than dimming the display and lock the buttons.

I would not say so... at least not for all types of pumps. With my one (the square shaped, grey from 2021) the power button is lighted in bright green until you really "power off". Dimming the display and locking the buttons keeps this button lighted all the time. I am also looking for a patch to have the power button controlled via the GUI.

**Reply by visualapproach** [reply]
Got it. New definition of power off is "dimming the screen exept for the green LED on some pumps" 😄 
May add feature in the future
